### PR TITLE
feat: clients enhancement

### PR DESCRIPTION
**Pull request by ekkinox**

### CODE REVIEW

File: src/Service/Score/Client/ScoreServiceClient.php [hunk 1]
```diff
@@ -56,7 +63,60 @@ public function __construct(
...
+                $exception->getCode(),
+                $exception
+            );
+        }
+    }
+
+    /**
      * @throws LtiExceptionInterface
      */
     public function publishScore(
```
Comment (wazelin) [inline src/Service/Score/Client/ScoreServiceClient.php:122]: I wonder if we should, perhaps, mark this method as `@deprecated` and suggest using `publishScoreForClaim()` instead, which performs all the necessary claim checks.
WDYT?

Comment (ekkinox) [reply src/Service/Score/Client/ScoreServiceClient.php:122]: imo it's not deprecated, we should still allow to use this method.

example use case: we want to publish async, and we dont have access to the claim of the payload but only the url.

Comment (wazelin) [reply src/Service/Score/Client/ScoreServiceClient.php:122]: Excellent point. I do hope it will be obvious for the end-application developers :)

Comment (ekkinox) [reply src/Service/Score/Client/ScoreServiceClient.php:122]: will make sure of this in the documentation update

**Review by xvzf — APPROVED** [review]

**Pull request merged**